scripts/setup_fullnode.sh

```diff
@@ -419,14 +419,26 @@ function main() {
         pushd ${IOTEX_HOME}
         mkdir data log etc > /dev/null 2>&1
         popd
+    fi
 
-        wantdownload=N
-        read -p "Do you prefer to start from a snapshot, Download the db file. [Y/N] (Default: N)? " wantdownload
-        if [ "${wantdownload}X" = "YX" ] || [ "${wantdownload}X" = "yX" ];then
-            # Download db file
-            donwloadBlockDataFile
+    wantdownload=Y
+    if [ "$_PLUGINS_"X = "gateway"X ];then
+        read -p "Do you prefer to start from a snapshot, This will overwrite existing data。Download the db file. [Y/N] (Default: Y)? " wantdownload
+        if [[ "$runversion" == "v1.1"* && "$version" == "v1.2"* ]];then
+            if [ "$wantdownload"X = "N"X ] || [ "$wantdownload"X = "n"X ];then
```

Comment:
is it possible to combine these 2 if?
if too complicated, it's fine to leave it like this